ProductConstructionService.Api.Tests/JobsProcessorStatusTests.cs

```diff
@@ -0,0 +1,164 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using FluentAssertions;
+using ProductConstructionService.Api.Queue;
+
+namespace ProductConstructionService.Api.Tests;
+
+public class JobsProcessorStatusTests
+{
+    [Test, CancelAfter(1000)]
+    public async Task JobsProcessorStatusNormalFlow()
+    {
+        JobsProcessorScopeManager scopeManager = new();
+
+        // When it starts, the processor is not working
+        scopeManager.State.Should().Be(JobsProcessorState.Stopped);
+
+        TaskCompletionSource tcs1 = new();
+        Thread t = new(() =>
+        {
+            using (scopeManager.BeginJobScopeWhenReady(CancellationToken.None))
+            {
+                tcs1.SetResult();
+            }
+        });
+        t.Start();
+
+        // Wait for the worker to start and get blocked
+        while (t.ThreadState != ThreadState.WaitSleepJoin) ;
+
+        // Start the service again
+        scopeManager.Start();
+
+        // Wait for the worker to enter the job
+        await tcs1.Task;
+
+        scopeManager.State.Should().Be(JobsProcessorState.Working);
+
+        // The JobProcessor is working now, it shouldn't block on anything
+        await Task.Factory.StartNew(() =>
+        {
+            using (scopeManager.BeginJobScopeWhenReady(CancellationToken.None)) { }
+        });
```

Comment:
Can't we just call `BeginJobScopeWhenReady` directly and see it returns?

Reply:
yup, makes sense